docs/faq/code-analysis/how-long-does-it-take-for-my-repository-to-be-analyzed.md

```diff
@@ -1,6 +1,13 @@
 # How long does it take for my repository to be analyzed?
 
-Codacy usually takes under 5 minutes to analyze your repository, however it may take longer as this depends on the size of your repository.
+Codacy usually takes under 5 minutes to analyze your repository. It may however take longer, depending on a number of factors:
+
+-   The size of your repository
+-   Whether it's the initial analysis of your repository
+-   The priority of your analysis request, with open source projects having lower priority
+-   The current load on Codacy's servers
+-   The time it takes your Git provider to call the webhooks on Codacy
+-   Whether Codacy is currently experiencing issues or outages
 
 Codacy relies on post-commit hooks sent by your Git provider to trigger the analysis after each push to the repository, so if your analysis is taking a lot of time to start [check that the Post-Commit Hook integration for your repository is enabled](../../repositories-configure/integrations/post-commit-hooks.md).
```

Comment:
These are already troubleshooting two of the above steps.

We could consider merging them and integrating them with more troubleshooting options, perhaps in a table.

Reply:
I'm not sure that I understood what you mean.

What are the steps exactly, and which page includes troubleshooting instructions? If you're referring to the page "Post-commit hooks", be aware that we should review and refactor that page first (see https://github.com/codacy/docs/issues/682).

Reply:
Exactly, I'm referring to the last paragraphs currently on the page, containing solutions that relate to two of our new points:
* the link to the "Post-commit hooks" page relates to "The time it takes your Git provider to call the webhooks on Codacy"
* the tip to check Codacy's status page relates to "Whether Codacy or your Git provider is currently experiencing issues or outages"

With our new list of potential reasons, these two action points may end up looking a bit out of context.

Reply:
> The current load on Codacy's servers: we can vaguely mention avoiding peak load times, but I want to confirm that we are OK disclosing this information first

I don't think it's reasonable to ask customers to "avoid" the peak time, but the information could still be useful just for setting expectations. Either way, let's [check first if it's OK to reveal this information or not](https://codacy.atlassian.net/browse/DOCS-407?focusedCommentId=53978).

Reply:
The consensus is that we shouldn't mention it ([thread](https://codacy.slack.com/archives/CAL88HZL6/p1674641194737809)). I removed the relevant TODO.